crates/api/src/arc.rs

```diff
@@ -0,0 +1,386 @@
+//! Kitsune2 arcs represent a range of hash locations on the DHT.
+//!
+//! An arc can exist in its own right and refer to a range of locations on the DHT.
+//! It can also be used in context, such as an agent. Where it represents the range of locations
+//! that agent is responsible for.
+
+use serde::{Deserialize, Serialize};
+
+/// A basic definition of a storage arc compatible with the concept of
+/// storage and querying of items in a store that fall within that arc.
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq, Default)]
+pub enum BasicArc {
+    /// No DHT locations are contained within this arc.
+    #[default]
+    Empty,
+    /// All DHT locations are contained within this arc.
+    Full,
+    /// A specific range of DHT locations are contained within this arc.
+    ///
+    /// The lower bound is inclusive, the upper bound is exclusive.
+    Literal(u32, u32),
```

Comment:
I don't like how this serializes. Is this a case where `serde(untagged)` would be acceptable? https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5fe9bcf2c03e3f5f2c41e2f6bc82be16

Reply:
I also vote for changing this variant name to `Arc` instead of Literal.

Reply:
Can do yes, it requires not having two fields with no values so it forces the Full to go